lib/dispatcher.js

```diff
@@ -14,6 +18,20 @@ class Dispatcher extends EventEmitter {
   destroy () {
     throw new Error('not implemented')
   }
+
+  intercept (...interceptors) {
+    let dispatcher = this
+    for (const interceptor of interceptors) {
+      if (typeof interceptor !== 'function') {
+        throw new Error('invalid interceptor')
+      }
+      dispatcher = interceptor(dispatcher) ?? dispatcher
```

Comment:
WHat do you mean?

Reply:
I'm already seeing issues stating that they tried to chain two or three dispatchers and something didn't work as expected; I'm referring that we should state in the docs that each _interceptor_ should return the dispatcher clearly, otherwise, it can lead to inconsistencies.

More a documentation thing